### ISSUE TITLE
99 problems: finish 

### ISSUE THREAD

**Issue by Chris00** [ocaml/v2.ocaml.org#54]
Finish the inclusion of the [99 problems](http://www.ic.unicamp.br/~meidanis/courses/mc336/2006s2/funcional/L-99_Ninety-Nine_Lisp_Problems.html) and their [solutions](https://github.com/VictorNicollet/99-Problems-OCaml/tree/master/31-40).

**Comment by Chris00** [comment]
I imported all solutions from https://github.com/VictorNicollet/99-Problems-OCaml

**Comment by Chris00** [comment]
Not closing this tough as all questions from http://www.ic.unicamp.br/~meidanis/courses/mc336/2006s2/funcional/L-99_Ninety-Nine_Lisp_Problems.html need to be imported...  Then solutions must be written.  Finally everything should be checked (I hope we can find volunteers for this, this is certainly fun to find better solutions on than those written).  Maybe we can add a “discussion” section to some solutions.

**Comment by agarwal** [comment]
> Maybe we can add a “discussion” section to some solutions.
> 
> That's the kind of nice features I think we could support with a more
> dynamic infrastructure. I suspect this level of contribution via a git repo
> is unlikely.

**Comment by Chris00** [comment]
That's certainly possible to add it in an ad-hoc way.  The question is how to do it well.  Maybe an inspiration can come from [Real World Haskell](http://book.realworldhaskell.org/read/) which allows readers to comment paragraphs.  A CSS class is enough to pinpoint tags with commenting enabled but the problem is how to keep track of the comments when the content evolves...).

**Comment by samoht** [comment]
I quite like http://disqus.com/ for comments. It let you add dynamic comments to static webpages quite easily.

**Comment by Chris00** [comment]
I have completed the import of the 99 problems.  Now the real work begin.  Check all the statements and write clean solutions to the one without.  I am sure some people cannot resist a challenge — one should point them to the page during the OUD talk!

**Comment by agarwal** [comment]
> point them to the page during the OUD talk!
> 
> I'm hoping we'll get to make a short announcement to update the community
> about the website. But otherwise I'll throw in a comment at the end of the
> Biocaml talk.

This is a great page! I'm glad you added it.

**Comment by Chris00** [comment]
Glad you like it — it required quite some work!

**Comment by agarwal** [comment]
I just took another look, and you really went all out! This page serves as
inspiration for the rest of the site.

**Comment by agarwal** [comment]
@Chris00 I consider the 99 problems page to be rather complete. If you agree, please close this issue.

**Comment by Chris00** [comment]
> @Chris00 I consider the 99 problems page to be rather complete. If you agree, please close this issue.

Yes but it is not complete.  I'd rather leave this open to make it obvious to potential contributors.

**Comment by dsheets** [comment]
I believe the contribution confusion was addressed by #553 and we are now getting contributions like #555, #556, and #557.

**Comment by sh4nth** [comment]
Can we please number the problems?  It makes it easier to navigate the page and remember where to continue from.  Also, I looked through the list and there are only 84 problems (and they seem to be related to gaps in the [original|http://www.ic.unicamp.br/~meidanis/courses/mc336/2006s2/funcional/L-99_Ninety-Nine_Lisp_Problems.html] itself.  Not sure if you want to maintain the gaps in the original or re-number the ones here?

**Comment by IniterWorker** [comment]
Hello, @Chris00, how I can contribute to this part? 

Because you quoted  [https://github.com/VictorNicollet/99-Problems-OCaml](https://github.com/VictorNicollet/99-Problems-OCaml), but the current staging on ocaml.org of 99 problems solutions is different. 

So I'm confused about it. What should I do to improve some solution, write it and contribute to your project?

**Comment by Chris00** [comment]
Hi,

On  2 January 2019 at 00:36 CET, Walter Bonetti wrote:
>
> Hello, @Chris00, how I can contribute to this part?
>
> Because you quoted https://github.com/VictorNicollet/99-Problems-OCaml, but the current staging on ocaml.org of 99 problems solutions is different.

Most of the solutions have not been copied from the above site.

> So I'm confused about it. What should I do to improve some solution, write it and contribute to your project?

Before improving solutions, it would be better to write some for questions that have not yet received any, making sure it is high quality code as it has to show good practices to readers.

Best,
C.
